capstone/src/main/webapp/statistics_script.js

```diff
@@ -23,41 +29,79 @@ async function getSurveyResponses() {
     '65+': 0,
     'Unknown': 0,
   };
+
+  const directExpCount = {
+    'Prefer not to answer': 0,
+    'Yes': 0,
+    'No': 0,
+  };
+
+  const completionCount = {
+    'Partial': 0,
+    'Complete': 0,
+  };
+
+  const responseTimeAverage = {
+    'Response One': 0,
+    'Response Two': 0,
+    'Response Three': 0,
+  };
+
+  /* Stores a respective field in each object of the list */
+  const scores = [];
   const genders = [];
-  const responseTimes = [];
+  const responseTimesThree = [];
   const ages = [];
   const days = [];
 
   for (let i = 0; i < list.length; i++) {
     scores.push(list[i].score);
-    responseTimes.push(list[i].responseTime);
+    responseTimesThree.push(list[i].responseTimeThree);
     genders.push(list[i].gender);
     ages.push(list[i].ageRange);
-    days.push(list[i].day);
+
+    const date = list[i].date;
+    const day = parseInt(date.substring(date.indexOf(' '), date.indexOf(',')));
```

Comment:
Why did you decide to do this analysis/count on the client side? I chose to do it the other way around and while its not exactly the same tradeoffs I'm curious as to what you view as the benefits/disavantages of the two approaches (generating frequencies of certain responses in client side JS versus server side when we redownload the results and run sentiment analysis on it.

Reply:
I wasn't too sure how to do it on the server side and be able to add it to the DOM. Currently, my servlet fetches from datastore and returns a serialized array list. If I were to do the counting on the server side, is it possible to send all that separate data to the script file?

Reply:
If so, I am not sure if there will be increased latency because I'd have to send more data from the servlet to the client side

Reply:
In this case, there probably isn't a huge difference and either approach would work.

However, some thoughts:
* There is often more consistency in resources when processing server-side vs across various clients with differing compute capacity.
* There can be a greater flexibility in slicing and dicing the data within the FE if more raw data is passed to the client.
* Server-side computation that doesn't vary across users can heavily leverage caching within the server to speed up responses.